### PR TITLE
impl(w1r3/java): basic benchmark

### PR DESCRIPTION
**Pull request by coryan**

### CODE REVIEW

File: w1r3/java/src/main/java/W1R3.java [hunk 1]
```diff
@@ -0,0 +1,404 @@
...
+    }
+
+    public static class Transport {
+        public Transport(String n, Storage c) {
+            name = n;
+            client = c;
+        }
+
+        public final String name;
+        public final Storage client;
```
Comment (BenWhitehead) [inline w1r3/java/src/main/java/W1R3.java:136]: Java convention is to declare fields before constructors

File: w1r3/java/src/main/java/W1R3.java [hunk 2]
```diff
@@ -0,0 +1,404 @@
...
+        public BlobId upload(Storage client, BlobInfo info, ByteBuffer input) throws Exception;
+
+        public String name();
+    }
+
+    private void worker(
+            Transport[] transports, Uploader[] uploaders,
+            byte[] randomData, int seed, String instance,
+            OpenTelemetrySdk otelSdk) {
+        var random = new Random(seed);
```
Comment (BenWhitehead) [inline w1r3/java/src/main/java/W1R3.java:149]: Not sure if important in your opinion, but the distribution of Random is not uniform. If we want it to be uniform we'll need to add another dependency and update the code in a later PR.

File: w1r3/java/src/main/java/W1R3.java [hunk 3]
```diff
@@ -0,0 +1,404 @@
...
+                .setMeterProvider(meterProvider)
+                .buildAndRegisterGlobal();
+    }
+
+	// We want millisecond-sized histogram buckets. Floating point arithmetic is
+	// famously tricky, and time arithmetic is also error prone. Avoid most of
+	// these problems by using `Duration` to do all the arithmetic, and only
+    // convert to `double` (and seconds) when needed.
+    private static List<Double> makeLatencyBoundaries() {
+        var boundaries = new ArrayList<Double>();
```
Comment (BenWhitehead) [inline w1r3/java/src/main/java/W1R3.java:365]: This is a list of durations in seconds?

File: w1r3/java/src/main/java/W1R3.java [hunk 4]
```diff
@@ -0,0 +1,404 @@
...
+        } catch (IOException e) {
+        }
+        return "unknown";
+    }
+
+    private static final String sdkVersion = getPackageVersion("com.google.cloud", "google-cloud-storage");
+    private static final String httpClientVersion = getPackageVersion("com.google.http-client", "google-http-client");
+    private static final String protobufVersion = getPackageVersion("com.google.protobuf", "protobuf-java");
+    private static final String grpcVersion = getPackageVersion("io.grpc", "grpc-core");
+};
```
Comment (BenWhitehead) [inline w1r3/java/src/main/java/W1R3.java:404]: No semicolon needed in java

**Review by BenWhitehead — CHANGES_REQUESTED** [review]

File: w1r3/java/src/main/java/W1R3.java [hunk 5]
```diff
@@ -0,0 +1,404 @@
...
+import java.util.Properties;
+import java.util.Random;
+import java.util.UUID;
+
+@Command(name = "w1r3", mixinStandardHelpOptions = true, version = "", description = "Runs the w1r3 benchmark.")
+final class W1R3 implements Callable<Integer> {
+    private static final String serviceName = "w1r3";
+    private static final String scopeName = "w1r3";
+    private static final String scopeVersion = "0.0.1";
+    private static final double sampleRate = 0.05;
```
[the remark this answers is not in the document]
Comment (coryan) [reply w1r3/java/src/main/java/W1R3.java]: Done

File: w1r3/java/src/main/java/W1R3.java [hunk 6]
```diff
@@ -0,0 +1,404 @@
...
+    private final int KB = 1000;
+    private final int KiB = 1024;
+    private final int MB = 1000 * KB;
+    private final int MiB = 1024 * KiB;
```
[the remark this answers is not in the document]
Comment (coryan) [reply w1r3/java/src/main/java/W1R3.java]: Done. I am not sure what is the convention some something like `KiB`.

File: w1r3/java/src/main/java/W1R3.java [hunk 7]
```diff
@@ -0,0 +1,404 @@
...
+        System.out.printf("# object-sizes: %s\n", Arrays.toString(this.objectSizes));
+        System.out.printf("# transports: %s\n", Arrays.toString(this.transports));
+        System.out.printf("# project-id: %s\n", this.project);
+        System.out.printf("# bucket: %s\n", this.bucket);
+        System.out.printf("# deployment: %s\n", this.deployment);
+        System.out.printf("# instance: %s\n", instance);
+        System.out.printf("# Java SDK Version %s\n", sdkVersion);
+        System.out.printf("# HTTP Version %s\n", httpClientVersion);
+        System.out.printf("# Protobuf Version %s\n", protobufVersion);
+        System.out.printf("# gRPC Version %s\n", grpcVersion);
```
[the remark this answers is not in the document]
Comment (coryan) [reply w1r3/java/src/main/java/W1R3.java]: Done.

File: w1r3/java/src/main/java/W1R3.java [hunk 8]
```diff
@@ -0,0 +1,404 @@
...
+
+        var random = new SecureRandom();
+        var randomData = makeRandomData(this.objectSizes, random);
+
+        var otelSdk = setupOpenTelemetrySdk(instance);
```
[the remark this answers is not in the document]
Comment (coryan) [reply w1r3/java/src/main/java/W1R3.java]: Done

File: w1r3/java/src/main/java/W1R3.java (hunk 1 above)
Comment (coryan) [reply w1r3/java/src/main/java/W1R3.java]: Done.

File: w1r3/java/src/main/java/W1R3.java (hunk 2 above)
Comment (coryan) [reply w1r3/java/src/main/java/W1R3.java:151]: Yeah, we should fix that in a separate pass. #64

File: w1r3/java/src/main/java/W1R3.java (hunk 3 above)
Comment (coryan) [reply w1r3/java/src/main/java/W1R3.java:368]: Yes, updated the comment.

File: w1r3/java/src/main/java/W1R3.java (hunk 4 above)
Comment (coryan) [reply w1r3/java/src/main/java/W1R3.java]: Removed.

File: w1r3/java/src/main/java/W1R3.java [hunk 9]
```diff
@@ -0,0 +1,404 @@
...
+            }
+        } catch (IOException e) {
+        }
+        return "unknown";
+    }
+
+    private static final String sdkVersion = getPackageVersion("com.google.cloud", "google-cloud-storage");
+    private static final String httpClientVersion = getPackageVersion("com.google.http-client", "google-http-client");
+    private static final String protobufVersion = getPackageVersion("com.google.protobuf", "protobuf-java");
+    private static final String grpcVersion = getPackageVersion("io.grpc", "grpc-core");
```
[the remark this answers is not in the document]
Comment (coryan) [reply w1r3/java/src/main/java/W1R3.java]: Done, also renamed them to match the style for constants (I hope).

**Review by BenWhitehead — APPROVED** [review]

**Pull request merged**